src/lib/proposer/proposer.ml

```diff
@@ -176,7 +178,20 @@ module Make (Inputs : Inputs_intf) :
               ( Ledger_builder_diff.With_valid_signatures_and_proofs.payments
                   diff
                 :> Payment.t list )
-            ~ledger:(Ledger_builder.ledger ledger_builder)
+            ~snarked_ledger_hash:(fun () ->
```

Comment:
instead of passing the `snarked_ledger_hash` implicitly through a closure, could you instead pass the `ledger_builder` directly to `generate_transition` (et al) and do this work over there? Using callbacks like this makes it hard to trace code.

Reply:
See #1092